internal/gatewayapi/runner/runner.go

```diff
@@ -76,10 +77,12 @@ func (r *Runner) subscribeAndTranslate(ctx context.Context) {
 			}
 			// Translate to IR
 			result := t.Translate(val)
-
 			var curKeys, newKeys []string
 			// Get current IR keys
-			for key := range r.InfraIR.LoadAll() {
+			filteredInfra := r.InfraIR.LoadAllMatching(func(_ string, ir *ir.Infra) bool {
```

Comment:
im a little wary of relying on labels to decide translation.here is one suggestion to solve this, please lmk what you think
* change the pub-sub watchable interface to publish a `providerResources` which looks like `map[string]*gatewayi.Resources` where the key is the gateway class name
* loop through the `providerResources` and for each gatewayClass, call `gatewayapi.Translate`
* remember all the IR values
* once done looping, clean up any old IRs
* publish all IRs

Reply:
Understood, I thought that it would be safe anyway to keep labels as they are used in IR only.
The problem was with dealing with the keys, runner already watches for changes in map[string]*gatewayi.Resources, where the key is a GatewayClass name and InfraIR has a key in the Gateway namespace/ Gateway name format or GatewayClass Name for merged gateways.

Reply:
not sure I understood the problem, can you rephrase that

Reply:
I understand your issue :) , the watchable message needs to be a map representing all the resources, so the key here needs to the controller name, but its value is also a map which is a list of tuples with key as gatewayclass name and val as child resources

Reply:
sorry, it was easy to get confused, I should rephrased that it wasn't that big of an issue, just the label approach was much simpler than matching all gateways from Infra keys :)

Reply:
ok, so you would like to change the interface completely and rework this update func, now I think I understand your idea
let me try this and I will get back with it

Reply:
thanks @cnvergence !

Reply:
hey @arkodg, you would like to have this watchable message look like?
```code
type ControlledResources struct {
	ControlledResources watchable.Map[string, *ProviderResources]
}

// ProviderResources message
type ProviderResources struct {
	// GatewayAPIResources is a map from a GatewayClass name to
	// a group of gateway API and other related resources.
	GatewayAPIResources watchable.Map[string, *gatewayapi.Resources]

	// GatewayAPIStatuses is a group of gateway api
	// resource statuses maps.
	GatewayAPIStatuses

	// PolicyStatuses is a group of policy statuses maps.
	PolicyStatuses
}
```
this seems to be a big refactor, so want to make sure

Reply:
```
type ControllerResources map[string]*gatewayapi.Resources

// ProviderResources message
type ProviderResources struct {
	// GatewayAPIResources is a map from a GatewayClass name to
	// a group of gateway API and other related resources.
	GatewayAPIResources watchable.Map[string, *ControllerResources]

	// GatewayAPIStatuses is a group of gateway api
	// resource statuses maps.
	GatewayAPIStatuses

	// PolicyStatuses is a group of policy statuses maps.
	PolicyStatuses
}
```

probably something more like this, wdyt ?

Reply:
That should do the work for the translator and matching old IR, I will check on how to use the new ProviderResources in the kubernetes provider.